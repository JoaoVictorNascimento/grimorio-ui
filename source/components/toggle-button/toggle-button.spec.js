import '../../../internals/test/helper';

import ToggleButton from './toggle-button-component';

/** @test {ToggleButton} */
describe('ToggleButton component', () => {
  /** @test {ToggleButton#render} */
  let wrapper
  const props = {
    onClick: jest.fn(),
    value: false
  }

  beforeAll(() => {
    wrapper = shallow(
      <ToggleButton {...props} />
    );
  })

  it('render correctly with default values', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('on Click YES button should return true', () => {
    jest.spyOn(props, 'onClick')
    wrapper.find('.toggleBoolLeft').simulate('click')
    expect(props.onClick).toHaveBeenCalledWith(true)
  })

  it('on Click NO button should return false', () => {
    jest.spyOn(props, 'onClick')
    wrapper.find('.toggleBoolRight').simulate('click')
    expect(props.onClick).toHaveBeenCalledWith(false)
  })
});
