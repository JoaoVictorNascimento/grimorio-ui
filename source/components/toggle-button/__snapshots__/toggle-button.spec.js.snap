// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton component render correctly with default values 1`] = `
"<div className=\\"toggleBool\\">
  <span onClick={[Function: onClick]} className=\\"toggleBoolLeft\\">
    Sim
  </span>
  <span onClick={[Function: onClick]} className=\\"toggleBoolRight active\\">
    NÃ£o
  </span>
</div>"
`;
